tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -191,6 +195,81 @@ def _predict_proba(self, X, **kwargs):
+        if not self.model.bootstrap:
+            raise ValueError('Forest models must set `bootstrap=True` to compute out-of-fold predictions via out-of-bag predictions.')
+
+        # TODO: This can also be done via setting `oob_score=True` in model params,
+        #  but getting the correct `pred_time_val` that way is not easy, since we can't time the internal call.
+        if (getattr(self.model, "oob_decision_function_", None) is None and getattr(self.model, "oob_prediction_", None) is None) \
+                and callable(getattr(self.model, "_set_oob_score", None)):
+            X = self.preprocess(X)
+
+            if getattr(self.model, "n_classes_", None) is not None:
+                if self.model.n_outputs_ == 1:
+                    self.model.n_classes_ = [self.model.n_classes_]
+            from sklearn.tree._tree import DTYPE, DOUBLE
+            X, y = self.model._validate_data(X, y, multi_output=True, accept_sparse="csc", dtype=DTYPE)
+            if y.ndim == 1:
+                # reshape is necessary to preserve the data contiguity against vs
+                # [:, np.newaxis] that does not.
+                y = np.reshape(y, (-1, 1))
+            if getattr(y, "dtype", None) != DOUBLE or not y.flags.contiguous:
+                y = np.ascontiguousarray(y, dtype=DOUBLE)
+            self.model._set_oob_score(X, y)
+            if getattr(self.model, "n_classes_", None) is not None:
+                if self.model.n_outputs_ == 1:
+                    self.model.n_classes_ = self.model.n_classes_[0]
+
+        if getattr(self.model, "oob_decision_function_", None) is not None:
+            y_oof_pred_proba = self.model.oob_decision_function_
+            self.model.oob_decision_function_ = None  # save memory
+        elif getattr(self.model, "oob_prediction_", None) is not None:
+            y_oof_pred_proba = self.model.oob_prediction_
+            self.model.oob_prediction_ = None  # save memory
+        else:
+            raise AssertionError(f'Model class {type(self.model)} does not support out-of-fold prediction generation.')
+
+        # TODO: Regression does not return NaN for missing rows, instead it sets them to 0. This makes life hard.
+        #  The below code corrects the missing rows to NaN instead of 0.
+        # Don't bother if >40 trees, near impossible to have missing
+        if self.problem_type == REGRESSION and self.model.n_estimators <= 40:
+            from sklearn.ensemble._forest import _get_n_samples_bootstrap, _generate_unsampled_indices
+            n_samples = len(y)
+
+            n_predictions = np.zeros(n_samples)
+            n_samples_bootstrap = _get_n_samples_bootstrap(n_samples, self.model.max_samples)
+            for estimator in self.model.estimators_:
+                unsampled_indices = _generate_unsampled_indices(estimator.random_state, n_samples, n_samples_bootstrap)
+                n_predictions[unsampled_indices] += 1
+            missing_row_mask = n_predictions == 0
```

Comment:
what about legit zeroes? For example you predict a temperature between -100 and 100?

Reply:
Another point: do we want exact 0 here? Float point errors can do this:
```
0.0003 - 0.0002 - 0.0001 == 0
```
This results in `False`